pkg/tsdb/cloudwatch/models/request_parser_test.go

```diff
@@ -429,175 +674,223 @@ func Test_migrateAliasToDynamicLabel_single_query_preserves_old_alias_and_create
+				inputLabelJson:                    `"label":"some label",`,
+				dynamicLabelsFeatureToggleEnabled: true,
+				expectedLabel:                     "some label"},
+			"when label does not exist, feature toggle is disabled": {
+				inputLabelJson:                    "",
+				dynamicLabelsFeatureToggleEnabled: false,
+				expectedLabel:                     "",
+			},
+			"when label already exists, feature toggle is disabled": {
+				inputLabelJson:                    `"label":"some label",`,
+				dynamicLabelsFeatureToggleEnabled: false,
+				expectedLabel:                     "some label"},
 		}
 
 		for name, tc := range testCases {
 			t.Run(name, func(t *testing.T) {
-				migratedQueries, err := migrateLegacyQuery(
-					[]backend.DataQuery{
-						{
-							RefID:     "A",
-							QueryType: "timeSeriesQuery",
-							JSON: []byte(fmt.Sprintf(`{
-					"region": "us-east-1",
-					"namespace": "ec2",
-					"metricName": "CPUUtilization",
-					"alias": "{{period}} {{any_other_word}}",
-					%s
-					"dimensions": {
-					  "InstanceId": ["test"]
+				query := []backend.DataQuery{
+					{
+						JSON: json.RawMessage(fmt.Sprintf(`{
+				   "refId":"A",
+				   "region":"us-east-1",
+				   "namespace":"ec2",
+				   "metricName":"CPUUtilization",
+				   "alias":"{{period}} {{any_other_word}}",
+				   %s
+				   "dimensions":{"InstanceId":["test"]},
+				   "statistic":"Average",
+				   "period":"600",
+				   "hide":false
+				}`, tc.inputLabelJson)),
 					},
-					"statistic": "Average",
-					"period": "600",
-					"hide": false
-				  }`, tc.labelJson))},
-					}, tc.dynamicLabelsFeatureToggleEnabled)
-				require.NoError(t, err)
-				require.Equal(t, 1, len(migratedQueries))
-
-				assert.JSONEq(t,
-					fmt.Sprintf(`{
-					   "alias":"{{period}} {{any_other_word}}",
-					   %s
-					   "dimensions":{
-						  "InstanceId":[
-							 "test"
-						  ]
-					   },
-					   "hide":false,
-					   "metricName":"CPUUtilization",
-					   "namespace":"ec2",
-					   "period":"600",
-					   "region":"us-east-1",
-					   "statistic":"Average"
-					}`, tc.labelJson),
-					string(migratedQueries[0].JSON))
+				}
+				res, err := ParseMetricDataQueries(query, time.Now(), time.Now(), tc.dynamicLabelsFeatureToggleEnabled)
+				assert.NoError(t, err)
+
+				require.Len(t, res, 1)
+				require.NotNil(t, res[0])
+
+				assert.Equal(t, "{{period}} {{any_other_word}}", res[0].Alias)
+				assert.Equal(t, tc.expectedLabel, res[0].Label)
+				assert.Equal(t, map[string][]string{"InstanceId": {"test"}}, res[0].Dimensions)
+				assert.Equal(t, true, res[0].ReturnData)
+				assert.Equal(t, "CPUUtilization", res[0].MetricName)
+				assert.Equal(t, "ec2", res[0].Namespace)
+				assert.Equal(t, 600, res[0].Period)
+				assert.Equal(t, "us-east-1", res[0].Region)
+				assert.Equal(t, "Average", res[0].Statistic)
 			})
 		}
 	})
 }
+
+
+func Test_ParseMetricDataQueries_statistics_and_query_type_validation_and_MatchExact_initialization(t *testing.T) {
```

Comment:
These are new tests for some validation/default value of MatchExact.

I can also put all new tests at the bottom of this file if that makes reviewing easier, please let me know.
Currently new tests were added nearer other related tests.

Reply:
This is perfectly fine!